Task2/Project/Shapes/Commands/Vertex/AddVertex.cs

```diff
@@ -13,21 +13,24 @@ public class AddVertex : Interfaces.ICommand
         // FIELDS
         Models.Canvas baseCanvas;
         Models.Vertex target;
+        Models.UndoRedoManager workCommandManger;
 
 
         //CONSTRUCTORS
         /// <summary>
-        /// Constructor with 2 parameters
+        /// Constructor with 3 parameters
         /// </summary>
         /// <param name="baseCanvas">Vertex in which will be added vertex</param>
         /// <param name="target">Added vertex</param>
+        /// <param name="workCommandManger">program command manager</param>
         /// <exception cref="System.NullReferenceException">Pentagon or vertex doesn't exist!</exception>
-        AddVertex(Models.Canvas baseCanvas, Models.Vertex target)
+        AddVertex(Models.Canvas baseCanvas, Models.Vertex target, Models.UndoRedoManager workCommandManger)
         {
             if (baseCanvas != null && target != null)

```

Comment:
UndoRedoManager should also be checked
also they should be checked separately and thrown 3 exception